### PR TITLE
Flesh out new vsetvl{i} proposal

### LINKED ISSUE #264: Change behavior of vsetvl{i} rd, x0, {imm,rs2} to use vl instead of ~0 as AVL
Currently, if rs1 of vsetvl{i} is x0, vl is set to VLMAX.  The proposal is to change behavior to have rs1=x0 indicate that current vl should be used as AVL.  In the case that SEW/LMUL is unchanged, this value of vl should also be unchanged.   This avoids having to keep AVL in a scalar register.

Obtaining VLMAX will now require a "li t1, ~0; vsetvli t0, t1, ex,my" sequence, but this is less frequently needed than switching SEW/LMUL with vl unchanged.

### PR DESCRIPTION
**Pull request by aswaterman**
Closes #264

### CODE REVIEW

File: v-spec.adoc [hunk 1]
```diff
@@ -1132,15 +1132,23 @@ the `vill` bit is set in `vtype`, the remaining bits in `vtype` are
...
 The requested application vector length (AVL) is passed in `rs1` as an
-unsigned integer.  Using `x0` as the `rs1` register specifier, encodes
-an infinite AVL, and so requests the maximum possible vector length.
+unsigned integer.  Using `x0` as the `rs1` register specifier requests
+an AVL equal to the current `vl`.
 
-NOTE: A `vsetvl{i}` with AVL=`x0` can be used to read current VLMAX,
-though this does overwrite `vl`.
+NOTE: Executing `vsetvl{i}` with `rs1`=`x0` is guaranteed to keep the
+current `vl` intact if VLMAX is not reduced.
```
Comment (solomatnikov) [inline v-spec.adoc]: `vl` is not reset and it can have arbitrary value before 1st `vsetvli`

Comment (aswaterman) [reply v-spec.adoc]: "The vector extension must have a consistent state at reset" -- so `vl` can be arbitrary only up to the point that it remains valid for the reset value of `vtype`, in which case this statement still holds.

Comment (aswaterman) [reply v-spec.adoc]: (And of course if `vtype.vill` is high out of reset, then `vl` must appear to be zero, in which case this statement holds, too.)

Comment (solomatnikov) [reply v-spec.adoc]: The note says it is only recommended to set `vl` to zero at reset

Comment (aswaterman) [reply v-spec.adoc]: Yeah, but as written, the statement "The vector extension must have a consistent state at reset" is a requirement.

**Pull request merged**